bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/Schedule.java

```diff
@@ -225,6 +227,74 @@ public Schedule(MiniModel model, ParamSet params, int upgradingFrom) {
         init(false, ImOption.empty());
     }
 
+    // An alloc is considered in range if it starts before the end of the block
+    // and ends after the start of the block.
+    private static boolean inRange(
+        final ParamSet alloc,
+        final long     start,
+        final long     end
+    ) {
+        final long aStart = Pio.getLongValue(alloc, "start", Long.MAX_VALUE);
+        final long aEnd   = Pio.getLongValue(alloc, "end",   Long.MIN_VALUE);
+        return (aStart <= aEnd) && (aStart <= end) && (aEnd >= start);
+    }
+
+    // An alloc is out of range if it is not inRange.
+    private static boolean outOfRange(
+        final ParamSet alloc,
+        final long     start,
+        final long     end
+    ) {
+        return !inRange(alloc, start, end);
+    }
+
+    /**
+     * Determines whether all allocs at least start or end within the block
+     * time range.  If any fall outside of the block time range, an IOException
+     * is thrown listing the offending allocs.
+     *
+     * @param core param set describing the schedule, its variants and the
+     *             allocs they contain
+     * @param blocks the schedule time block(s)
+     *
+     * @throws IOException if any allocation falls completely outside of the
+     * range of the time blocks
+     */
+    public static void validateAllocs(
+        final ParamSet core,
+        final BlockUnion blocks
+    ) throws IOException {
+        final SortedSet<Block> intervals = blocks.getIntervals();
+        final long start = intervals.first().getStart();
+        final long end   = intervals.last().getEnd();
+
+        // A listing of the observation ids that fall out of range, if any.
+        final ImList<String> outOfRangeObsList =
+            ImOption.apply(core.getParamSet(PROP_VARIANTS))
+                .toImList()
+                .flatMap(vs -> DefaultImList.create(vs.getParamSets(VariantList.PROP_MEMBER)))
+                .flatMap(v  -> DefaultImList.create(v.getParamSets(Variant.PROP_ALLOCS)))
+                .flatMap(as -> DefaultImList.create(as.getParamSets(AllocSet.PROP_MEMBER)))
+                .filter(a -> outOfRange(a, start, end))
+                .map(a -> Pio.getValue(a, "obs"));
+
+        if (outOfRangeObsList.nonEmpty()) {
+            // Trim duplicates.
+            final ImList<String> os = DefaultImList.create(new TreeSet<>(outOfRangeObsList.toList()));
+
+            // Show the problem in the exception message, which ends up being
+            // displayed to the user from the open dialogs.
+            throw new IOException(
+                os.take(10).mkString(
+                    "Some observations are out of the schedule's time range:\n",
```

Comment:
will you display this message?

Reply:
Yeah, it is [displayed in the open dialog](https://github.com/gemini-hlsw/ocs/blob/develop/bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/ui/action/AbstractOpenAction.java#L118-L120).